saw-central/src/SAWCentral/Crucible/Common/Pred.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module SAWCentral.Crucible.Common.Pred (
+    checkBooleanType
+    ) where
+
+import Cryptol.TypeCheck.Type (tIsBit)
+import CryptolSAWCore.TypedTerm (mkTypedTerm, ttType, ttIsMono)
+import SAWCore.SharedTerm (SharedContext, Term)
+
+-- | Ensure that the term has Cryptol type `Bit`
+checkBooleanType :: SharedContext -> Term -> IO ()
+checkBooleanType sc tm = do
+  tt <- mkTypedTerm sc tm
+  case ttIsMono (ttType tt) of
+    Just ty | tIsBit ty -> pure ()
+            | otherwise -> fail $ "Expected type Bit, got: " ++ show ty
```

Comment:
Relatedly, I think it would improve readability to line up the expected and actual types. Something like:

```suggestion
            | otherwise -> fail $ unlines
                [ "Expected type: Bit"
                , "Actual type:   " ++ show (ppTerm PrecNone ty)
                ]
```